### ISSUE TITLE
Pyright type stubs fail for google.cloud.storage package

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#2393]
**Describe the bug**
Both generation and use of type stubs for Google Cloud packages do not work.

**To Reproduce**
Install google-cloud-storage: `pip install google-cloud-storage`.
Generate type stubs `pyright --createstub google.cloud.storage`

Instead of stubs being generated at `./typings/google/cloud/storage/...` they're added to `./typings/google/storage/...`. Additionally, even when manually relocating the stubs to `./typings/google/cloud/storage/...`, pyright reports errors for the stub contents.

For example, the import `from google.cloud import storage` resolves (sort of), but when trying to use `storage.Client('...')` an error is reported. 

Additionally, running the pyright CLI tool generates errors:

```
  /Users/schapman/workspace/.../my_scrupt.py:67:22 - error: "Client" is not a known member of module (reportGeneralTypeIssues)
```

**Expected behavior**
Generated type stubs should be generated in a way that resolves the type lookups correctly.

**Screenshots or Code**
VS Code errors:

![image](https://user-images.githubusercontent.com/3920249/136390516-3fd1d137-f07d-4329-9f48-d7a032abfc80.png)

![image](https://user-images.githubusercontent.com/3920249/136390553-2ad4c26e-91b1-4294-9400-2bc59cfea12f.png)

The generated stub in `./typings/google/storage/__init__.pyi` has `from google.cloud.storage.client import Client`

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.
I am running both as an extension (through Pylance v2021.10.0) and the CLI tool directly:

```sh
➤ pyright --version
pyright 1.1.176
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. We weren't properly handling module paths with more than one dot, such as "google.cloud.storage". I've fixed the bug and added new unit tests to cover this case.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.177, which I just published. It will also be included in the next release of pylance.
